src/main/scala/org/broadinstitute/hail/sparkextras/OrderedRDD.scala

```diff
@@ -85,19 +103,23 @@ object OrderedRDD {
     } else {
       info("Ordering unsorted dataset with network shuffle")
       val ranges = calculateKeyRanges(keys.map(kOk.project))
-      val partitioner = OrderedPartitioner(ranges)
-      OrderedRDD[PK, K, V](new ShuffledRDD[K, V, V](rdd, partitioner).setKeyOrdering(kOk.kOrd), partitioner)
+      apply(rdd, ranges)
     }
   }
 
-
   def apply[PK, K, V](rdd: RDD[(K, V)],
     orderedPartitioner: OrderedPartitioner[PK, K])(implicit kOk: OrderedKey[PK, K]): OrderedRDD[PK, K, V] = {
     import Ordering.Implicits._
     import kOk.pkOrd
     import kOk.kOrd
     import kOk.pkct
 
+    rdd.partitioner match {
+      /* if we verified rdd is K-sorted, it won't necessarily be partitioned */
+      case Some(p) => assert(p eq orderedPartitioner)
```

Comment:
why this assertion?  An RDD could come in as RangePartitioned, which would totally work for coercion